reframe/frontend/cli.py

```diff
@@ -640,11 +659,22 @@ def restrict_logging():
         argparser.print_help()
         sys.exit(1)
 
+    autodetect_opts = {
+        'autodetect_method': options.autodetect_method or 'hostname',
+        'autodetect_xthostname': (options.autodetect_xthostname
+                                  if options.autodetect_xthostname is not None
+                                  else True),
+        'autodetect_fqdn': (options.autodetect_fqdn
+                            if options.autodetect_fqdn is not None
+                            else True),
+    }
+
     # First configure logging with our generic configuration so as to be able
     # to print pretty messages; logging will be reconfigured by user's
     # configuration later
     site_config = config.load_config(
-        os.path.join(reframe.INSTALL_PREFIX, 'reframe/core/settings.py')
+        os.path.join(reframe.INSTALL_PREFIX, 'reframe/core/settings.py'),
+        autodetect_opts=autodetect_opts
```

Comment:
I would leave as before since with `None` the method `load_config` will get the defaults anyway.
```suggestion
        os.path.join(reframe.INSTALL_PREFIX, 'reframe/core/settings.py')
```